File: nim/day01/solve.nim
import std/[strformat, strutils, sequtils]

func countIncreasing(nums: openArray[int], step: int): int =
  for i in step .. nums.high:
    if nums[i] > nums[i - step]:
      inc result

proc solve* =
  let depths = ".input/day01".readFile.strip.split.map parseInt
  let part1 = depths.countIncreasing 1
  let part2 = depths.countIncreasing 3
  echo &"day01: {part1} {part2}"

File: kotlin/day04/solve.kt
package aoc.day04

fun solve() {
  val input = java.io.File(".input", "day04").readText()
  val bingo = input.toBingo()
  val part1 = bingo.part1()
  val part2 = bingo.part2()
  println("day04: $part1 $part2")
}
